pkg/cookies/csrf_test.go

```diff
@@ -0,0 +1,166 @@
+package cookies
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/encryption"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("CSRF Cookie Tests", func() {
+	var (
+		csrf       *CSRF
+		cookieOpts *options.Cookie
+	)
+
+	BeforeEach(func() {
+		cookieOpts = &options.Cookie{
+			Name:     cookieName,
+			Secret:   cookieSecret,
+			Domains:  []string{cookieDomain},
+			Path:     cookiePath,
+			Expire:   time.Hour,
+			Secure:   true,
+			HTTPOnly: true,
+		}
+
+		var err error
+		csrf, err = NewCSRF(cookieOpts)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("NewCSRF", func() {
+		It("makes unique nonces", func() {
```

Comment:
Maybe we should check that if we create a second csrf that it is also different?